### PR TITLE
Exodii Ceramics

### PR DESCRIPTION
**Pull request by Drew4484**
#### Summary
Content "Exodii Ceramics"

#### Purpose of change
Exodii have encountered diverse technological bases in their travels, but we don't have a ton of examples of those other worlds. This adds some content based around a world the Exodii visited that had advanced ceramic production.

#### Describe the solution
Adds a new ceramic material used by the Exodii for a bulky but strong form of ceramics. Able to be reproduced from local materials and a carbon nanotube production process they discovered in a past jump, this is a bulky but tough material that can be used for armor plating and other high-wear applications like gun barrels. Also updates Exodii hardware to use tempered steel.

- [x] Add Exodii ceramic material
- [x] Add ceramic to Exodii quad armor
- [x] Add a large standing kiln for working larger ceramic pieces
- [x] Add a patch kit that can be used to repair damaged ceramic, sold by the Exodii merchant
- [x] Add one of the new kilns to the Exodii Castle
- [x] Update crafting recipe or make uncraftable

#### Describe alternatives you've considered
None

#### Testing
Created new world, character, proceeded to Exodii base. Verified edited terrain shows correctly. Verified kiln cannot be used for anything at this time. Spawned modified Exodii equipment to verify updated materials. 
image

#### Additional context
The added ceramic and change from 'steel' to 'tempered steel' makes Exodii tougher, but not dramatically.

### CODE REVIEW

**Comment by kevingranade** [comment]
I feel like the most obvious point of distinction for ceramics is lack of repairability. We're already way too permissive about what we allow to be repaired, and extending it to ceramics is a further escalation. Especially if you're talking about ballistics, the damage model isn't missing chips,  it's stress fractures throughout the piece. 

I'm also not clear on the gun barrel link,  we already don't really model barrel erosion, so what would be distinctive about ceramics in this use case?

**Comment by Kamejeir** [comment]
If this is mostly flavor and a reason to explain how the exodii armor their fellows somewhat durably without overloading them with steel, cool! Is there more to it?

**Comment by Drew4484** [comment]
> I feel like the most obvious point of distinction for ceramics is lack of repairability. We're already way too permissive about what we allow to be repaired, and extending it to ceramics is a further escalation. Especially if you're talking about ballistics, the damage model isn't missing chips, it's stress fractures throughout the piece.
> 
> I'm also not clear on the gun barrel link, we already don't really model barrel erosion, so what would be distinctive about ceramics in this use case?

The initial idea is that chipped armor can be patched but cracked armor has to be basically mulched, the nanotubes recovered, and recast. Patching can be done with a kiln, otherwise the player effectively would cash in the damaged plate towards the cost of a new one. When it gets to the point of player equipment, intact ceramic armor can non_functional into chipped ceramic armor which then non_functionals into shattered ceramic amor, with decreased defense for the first and effectively zero defense at shattered like our ballistic plates. Gun barrel is just flavor at this point, I don't expect to add any ceramic firearm components.

**Comment by Drew4484** [comment]
> If this is mostly flavor and a reason to explain how the exodii armor their fellows somewhat durably without overloading them with steel, cool! Is there more to it?

I don't plan to add a ceramic plate full body suit anytime soon, but more Exodii-sold ceramic items would be fun to explore.

**Comment by ampersand55** [comment]
One idea for a ceramic-type armor is to have something like a scale mail with the scales made out of ceramic material that can only be repaired by replacing the damaged scales. 

Same principle as the dragon skin armor, which was also "bulky but strong". 

https://en.wikipedia.org/wiki/Dragon_Skin
https://cdda-guide.nornagon.net/item/dragonskin

**Comment by Drew4484** [comment]
> One idea for a ceramic-type armor is to have something like a scale mail with the scales made out of ceramic material that can only be repaired by replacing the damaged scales.
> 
> Same principle as the dragon skin armor, which was also "bulky but strong".
> 
> https://en.wikipedia.org/wiki/Dragon_Skin https://cdda-guide.nornagon.net/item/dragonskin

The trick is that the Exodii don't really need body armor, they would directly attach it to their mechanical body. They stock a few interesting items they've collected, but it would take some significant work to get them to **produce** something like that for the survivor. Not ruling it out, but that's outside the scope of what I'm planning here.

**Comment by IdleSol** [comment]
>One idea for a ceramic-type armor is to have something like a scale mail with the scales made out of ceramic material that can only be repaired by replacing the damaged scales.

It's easy enough to do. Using the torso as an example. Create torso-covering armor from any fabric with any covering. Add N pockets: `"ablative": true,`.

We create one plate with a coverage equal to 100/N. (For example: N = 20). And that's where the implementation feature comes in. When taking protection into account, the coverage for one part of the body is summed up from all plates.

That is, as long as all N plates are unbroken, the coverage will be equal to 100%. But if one of the plates is broken, the coverage will be equal to `100 * ( N - 1 ) / N`. Consequently, the more plates destroyed, the more weaknesses in the armor.

The problem is that armor that covers more than one body part turns into a long list of pockets.

An alternative would be one pocket for ablative armor. But the plate, when damaged, is replaced by the exact same plate, but with less coverage. So there was 100% coverage, damage it and you get 95%. And continue to 0% or in two or three steps say the armor is destroyed and replace it with air.

**Comment by Drew4484** [comment]
> >One idea for a ceramic-type armor is to have something like a scale mail with the scales made out of ceramic material that can only be repaired by replacing the damaged scales.
> 
> It's easy enough to do. Using the torso as an example. Create torso-covering armor from any fabric with any covering. Add N pockets: `"ablative": true,`.
> 
> We create one plate with a coverage equal to 100/N. (For example: N = 20). And that's where the implementation feature comes in. When taking protection into account, the coverage for one part of the body is summed up from all plates.
> 
> That is, as long as all N plates are unbroken, the coverage will be equal to 100%. But if one of the plates is broken, the coverage will be equal to `100 * ( N - 1 ) / N`. Consequently, the more plates destroyed, the more weaknesses in the armor.
> 
> The problem is that armor that covers more than one body part turns into a long list of pockets.
> 
> An alternative would be one pocket for ablative armor. But the plate, when damaged, is replaced by the exact same plate, but with less coverage. So there was 100% coverage, damage it and you get 95%. And continue to 0% or in two or three steps say the armor is destroyed and replace it with air.

Oh yeah technical implementation isn't hard, it's the "convince the Exodii to make it" issue. As their main combat units, the Quads, can't wear personal armor, they might have a few pieces in their stockpiles but would be unlikely to help manufacture a piece of armor for a human. I'm fine with adding a vest for sale though.

**Comment by IdleSol** [comment]
Convince in terms of lore? Or how to technically implement it?

**Comment by ampersand55** [comment]
> The trick is that the Exodii don't really need body armor, they would directly attach it to their mechanical body.

I wasn't necessarily talking about a wearable body armor, I meant that the armor-plates could overlap like scales and be designed to be easily replaced, to get around the issue with ceramic being unrepairable.

**Comment by ampersand55** [comment]
I know the intent of this PR is to provide "examples of those other worlds" which I very much like. But if we just want a suitable ceramic-like armor material for the exodii I have a suggestion. 

Given these criteria:

1. It should be very heavy, making it ideal for super strong exodii to use but less so for us humans.
2. It should be able to be produced in bulk by the exodii themselves without nanotechnology. 
3. It should be just out of technological reach for humans but still within human understanding and usable and modifiable to a degree.
4. A survivor could be able to make it themselves with help of exodii equipment and materials. 
5. It should be ceramic-like and fit the rugged retro-futurist "vibe" of the exodii. 

I would suggest a compound between a hard/super dense transition metal and boron, carbon or nitrogen. These are known to format extremely strong covalent bonds in cubic or hexagonal crystal systems and could have a hardness approaching that of diamond.

These can be at modern tech level:

Some real-life such compounds that are extremely hard and heavy is tungsten boride, tungsten carbide, osmium diboride and rhenium diboride, which a density about twice that of steel. We already have tungsten carbide in the game in the form of "Layered Carbide" that's used in the phase immersion suit and the RM13 combat armor. 

These can be higher than modern tech level:

The making of similar such compounds is a very active area of research in material science, and it's possible that a [superhard material](https://en.wikipedia.org/wiki/Superhard_material) harder than diamond can exist. [Recently a tungsten nitride was designed using machine learning](https://www.sciencedirect.com/science/article/abs/pii/S2095927318302494).

These can be lower than modern tech level:

Rhenium diboride in particular was synthesized in the 60s, but it could have been made even earlier provided you had access to high-purity rhenium or boron powder. Essentially you would only need an electric arc furnace (invented in the 1800s), a way  to purge its atmosphere with an inert gas like helium or argon to prevent oxidation, and a controlled cooling system.

**Comment by Drew4484** [comment]
> I know the intent of this PR is to provide "examples of those other worlds" which I very much like. But if we just want a suitable ceramic-like armor material for the exodii I have a suggestion.
> 
> Given these criteria:
> 
> 1. It should be very heavy, making it ideal for super strong exodii to use but less so for us humans.
> 2. It should be able to be produced in bulk by the exodii themselves without nanotechnology.
> 3. It should be just out of technological reach for humans but still within human understanding and usable and modifiable to a degree.
> 4. A survivor could be able to make it themselves with help of exodii equipment and materials.
> 5. It should be ceramic-like and fit the rugged retro-futurist "vibe" of the exodii.
> 
> I would suggest a compound between a hard/super dense transition metal and boron, carbon or nitrogen. These are known to format extremely strong covalent bonds in cubic or hexagonal crystal systems and could have a hardness approaching that of diamond.
> 
> These can be at modern tech level:
> 
> Some real-life such compounds that are extremely hard and heavy is tungsten boride, tungsten carbide, osmium diboride and rhenium diboride, which a density about twice that of steel. We already have tungsten carbide in the game in the form of "Layered Carbide" that's used in the phase immersion suit and the RM13 combat armor.
> 
> These can be higher than modern tech level:
> 
> The making of similar such compounds is a very active area of research in material science, and it's possible that a [superhard material](https://en.wikipedia.org/wiki/Superhard_material) harder than diamond can exist. [Recently a tungsten nitride was designed using machine learning](https://www.sciencedirect.com/science/article/abs/pii/S2095927318302494).
> 
> These can be lower than modern tech level:
> 
> Rhenium diboride in particular was synthesized in the 60s, but it could have been made even earlier provided you had access to high-purity rhenium or boron powder. Essentially you would only need an electric arc furnace (invented in the 1800s), a way to purge its atmosphere with an inert gas like helium or argon to prevent oxidation, and a controlled cooling system.

See the comment and linked article in https://github.com/CleverRaven/Cataclysm-DDA/blob/2b787458df6946b7a659dace78bcdab71be1bf7d/data/json/materials.json#L1997
1. It is heavy, but less dense than steel, resulting in thicker bulky plates. This doesn't impact the Quads, but makes for bulkier armor with some loss of mobility at the joints (higher encumbrance than equivalent steel armor)
2. The Exodii's specific contribution to this is the carbon nanotube production. The output combines two different technologies they encountered, a world with common use of ceramic, and a world with an efficient CNT production process that does not require an entire factory complex. Since they use it for their Quads, they produce large homogenous slabs of plating with relative ease.
3. The ceramic base is something Earth could produce, but the closest equivalent to the CNT integration (in non-layered plating and capable of being produced at scale) is suspending small granules of material within the plate, similar to organically-formed nacre. Earth's material of choice is steel, and outside of specialized applications, we just don't build our world around ceramic. As far as modifying/repurposing, that comes back to convincing the Exodii to work with you to assist in the design and fabrication. Without the pre-cataclysm technological base, advanced ceramics are effectively unreproducible for the survivor, and mass CNT production is still years away for Earth. Nothing says existing plating can't be repurposed, and I intend to add an option for the exosuits (see #77309) to mount Exodii ceramic plating. Obtaining it will be a whole different issue. Keep in mind that the Exodii have encountered significantly more advanced technology than what we see them use day to day. The majority of it was passed over for practicality issues, they aren't going to equip Quads with a devastating laser weapon if that weapon requires exotic gasses they can't easily obtain. 
4. Outside of convincing the Exodii to help design and manufacture it, player design and production is basically a non-starter. There's a reason why most of the top-tier craftable armors are tied behind books, it takes *serious* research and knowledge to understand how to use a material like this effectively, short of just slapping sheets of it on and hoping for the best. Part of the reason for steel's ubiquitousness is that we understand steel's structure very well, having hundreds of years of research and application. Survivor Sharon isn't going to be able to do ballistic and kinetic testing to maximize the value of it, so why not stick with the steel they can already make that is *nearly* as good?
5. It is ceramic-like and fits the rugged retro-futurist "vibe' of the Exodii.

**Comment by Drew4484** [comment]
> > The trick is that the Exodii don't really need body armor, they would directly attach it to their mechanical body.
> 
> I wasn't necessarily talking about a wearable body armor, I meant that the armor-plates could overlap like scales and be designed to be easily replaced, to get around the issue with ceramic being unrepairable.

Small scale ceramic lamellar like the dragonskin definitely has a use! One thought I had with this is that the Exodii would use smaller hexagon plates around joints, so it isn't impossible to repurpose a batch of those by attaching them to a vest of some sort. They'd be much more willing to sell you a crate of something they can whip up in an afternoon than do the work of making a vest for you.

**Comment by kevingranade** [comment]
Only a tangental point,  but I feel like "convince faction to do full on research and development for you" is being treated as too viable an option, as in it defaults to "hell no" and would need an overwhelmingly good use case to overcome that.  

No one in the setting has enough spare resources to embark on real research and design unless it's a critical survival issue for the faction itself.

**Comment by ADekema** [comment]
Could a survivor perhaps do their own manufacturing of personal armour and balistic plates from larger pieces? You are working with entire plates from quads afterall. Wouldn´t it be possible to just cut it into pieces with the right powertools and use those pieces for your own lamellar or scale armours or perhaps just cut out a piece in the rough shape of a balistic plate or breastplate? 

Also gives the Exodii a means to ´sell´ you some of their own ceramic. They just sell you a few of thier surplus or perhapse slightly damaged armores but you need to do the manufacturing yourself.

**Comment by Drew4484** [comment]
> Could a survivor perhaps do their own manufacturing of personal armour and balistic plates from larger pieces? You are working with entire plates from quads afterall. Wouldn´t it be possible to just cut it into pieces with the right powertools and use those pieces for your own lamellar or scale armours or perhaps just cut out a piece in the rough shape of a balistic plate or breastplate? 
> 
> Also gives the Exodii a means to ´sell´ you some of their own ceramic. They just sell you a few of thier surplus or perhapse slightly damaged armores but you need to do the manufacturing yourself.

TL;DR: I'm not adding any player armor with this PR. The Exodii have no reason to make it for you, the Earth factions that could potentially do something with it lack motivation to shift from well-understood steel to barely-understood-and-not-dramatically-superior Exodii ceramics. I'm adding this to make the Exodii interesting, not to upend the armor meta again.

I think it would be fun to add some additional uses for Exodii ceramics but as was said above, the Exodii don't have much motivation to design armor for the player. Since the ceramic isn't from a single world but rather the combination of two different technologies, they don't have a stockpile of ceramic suits to sell you. Using small pieces for lames is possible, but cutting ceramics to fit would be an extremely tedious process requiring diamond or boron carbide cutting tools and the ability to measure stress along the surfaces and substratum. The (C:DDA's) Earth technology base to do that kind of work and the requisite testing to maximize effectiveness for something like fitted body armor is effectively gone. A survivor plated coat would be possible using smaller sections used to armor joints or somehow managing to cut pieces to size, however with ceramic armor you generally need plates to be over certain minimum dimensions so avoid a strike damaging the outer edge and compromising the plate sooner than a hit on the center.

File: data/json/monsters/cyborgs.json [hunk 1]
```diff
@@ -260,7 +260,7 @@
     "diff": 8,
     "volume": "120 L",
     "weight": "200 kg",
-    "species": [ "CYBORG", "ZOMBIE" ],
+    "species": [ "CYBORG", "ZOMBIE", "exodii_ceramic" ],
```
Comment (Procyonae) [inline data/json/monsters/cyborgs.json:263]: Think you wanted 3 lines down c;
